### ISSUE TITLE
Broaden scope of suppressed warnings for listings without a main function?

### ISSUE THREAD

**Issue by englishm** [rust-lang/mdBook#1192]
At present, code listings without a `main` function will be wrapped in one and annotated with an `allow` lint check attribute provided by the following code:

https://github.com/rust-lang/mdBook/blob/769cc0a7c14fe133fc5eb223f435473730d0086e/src/renderer/html_handlebars/hbs_renderer.rs#L636

Should this lint check attribute be broadened in scope to suppress warnings for things like `dead_code` as well?

There appears to be a `#![allow(unused)]` that may better fit the intent of this code.

---

Further background on how I arrived at this question is in Discord here:
https://discordapp.com/channels/442252698964721669/443492145567891458/700226963993919529

TL;DR: This auto-generated wrapper and attribute were visible to me because I was viewing a local copy of ["The Book"](https://doc.rust-lang.org/book/) in Emacs' [`eww`](https://www.gnu.org/software/emacs/manual/html_node/eww/index.html) text-mode browser. I noticed that the attribute didn't actually suppress the one warning about `dead_code` that the listing in Ch 3.1 introducing `const` generated.

I'm happy to create the small PR to change this auto-generated wrapper snippet from using the `#![allow(unused_variables)]` attribute to using the `#![allow(unused)]` attribute, but I first wanted to see if that's actually a desirable change.

**Comment by ehuss** [comment]
Closed by #1195.
